app/web/features/search/SearchResultsList.tsx

```diff
@@ -83,163 +75,107 @@ const useStyles = makeStyles((theme) => ({
-        } else {
-          map.current?.once("load", setFilter);
-        }
-      },
-    }
-  );
-  const isSearching = Object.keys(searchFilters.active).length !== 0;
+  const hasAtLeastOnePageResults = true;
 
   return (
     <Paper className={classes.mapResults}>
-      {error && <Alert severity="error">{error.message}</Alert>}
+      {error && <Alert severity="error">{error}</Alert>}
+
       <Hidden smDown>
         <SearchBox
-          searchFilters={searchFilters}
-          updateMapBoundingBox={updateMapBoundingBox}
+          searchType={searchType}
+          setSearchType={setSearchType}
+          locationResult={locationResult}
+          setLocationResult={setLocationResult}
+          setQueryName={setQueryName}
+          queryName={queryName}
         />
       </Hidden>
-      {isSearching ? (
-        isLoading ? (
-          <CircularProgress className={classes.baseMargin} />
-        ) : hasAtLeastOnePage(results, "resultsList") ? (
+
+      <>
+        {isLoading && <CircularProgress className={classes.baseMargin} />}
+
+        {!isLoading && !hasAtLeastOnePageResults && (
+          <TextBody className={classes.baseMargin}>
+            {t("search_result.no_user_result_message")}
+          </TextBody>
+        )}
+
+        {!isLoading && hasAtLeastOnePageResults && (
           <HorizontalScroller
             breakpoint="sm"
             className={classes.scroller}
-            isFetching={isFetching}
+            isFetching={isLoading}
             fetchNext={fetchNextPage}
-            hasMore={hasNextPage}
+            hasMore={hasNext}
           >
-            {results.pages
-              .flatMap((page) => page.resultsList)
-              .map((result) =>
-                result.user ? (
-                  <SearchResult
-                    id={`search-result-${result.user.userId}`}
-                    className={classes.searchResult}
-                    key={result.user.userId}
-                    user={result.user}
-                    onSelect={handleResultClick}
-                    highlight={result.user.userId === selectedResult}
-                  />
-                ) : null
-              )}
+            {results &&
+              results.pages
+                .flatMap((page) => page.resultsList)
+                .map((result) =>
+                  result.user ? (
+                    <SearchResult
+                      id={`search-result-${result.user.userId}`}
+                      className={classes.searchResult}
+                      key={result.user.userId}
+                      user={result.user}
+                      onSelect={() => {
+                        setSelectedResult({
+                          username: result.user?.username as string,
+                          userId: result.user?.userId as number,
+                          lng: result.user?.lng as number,
+                          lat: result.user?.lat as number,
+                        });
```

Comment:
Ideally add this type to the `useState` variable instead. This is telling it what type it is.

For example `'hello' as number` is a type assertion and will tell Typescript this is a number even though it's a string. So it overrides the type basically. You want to add the type to the source like so:

```typescript
const [count, setCount] = useState<number>(0);
```

That will give you the type protection you want.

Reply:
true.... will update it now :)

Reply:
this is the thing (sorry, I'm a bit hater of typescript myself)
![image](https://github.com/user-attachments/assets/97ac18eb-1784-433d-b871-ffc9cd8bb3c3)


as you can see it can't be undefined but it complains, that's why I added the `as string` thing

Reply:
I mean, selectedResult has the type I need, it's `result.user` here what creates this (unless I'm missing something I dont' see...)

Reply:
Try this:

```typescript
results?.pages
              .flatMap((page) => page.resultsList)
              .filter((result) => result.user) // filter out any without a user
              .map((result) => (
                <SearchResult
                  id={`search-result-${result.user!.userId}`} // add exclamation point bang ! to tell it user is definitely there
                  className={classes.searchResult}
                  key={result.user!.userId} // for all of them, only do this when you are sure
                  user={result.user!} // we are sure because we filtered out the ones without it above
                  onSelect={() => {
                    setSelectedResult({
                      username: result.user!.username,
                      userId: result.user!.userId,
                      lng: result.user!.lng,
                      lat: result.user!.lat,
                    });
                  }}
                  highlight={
                    selectedResult &&
                    result.user!.userId === selectedResult.userId
                  }
                />
              ))
```

Reply:
brilliant! 🤩 I didn't know about the exclamation syntax in TS (I'm a but hater of TS you know) also very good idea the filtering after the flatmap... it's implemented now! ✅